analyzers/src/SonarAnalyzer.CSharp/Rules/ObsoleteAttributes.cs

```diff
@@ -21,7 +21,15 @@
 namespace SonarAnalyzer.Rules.CSharp;
 
 [DiagnosticAnalyzer(LanguageNames.CSharp)]
-public sealed class RemoveObsoleteCode : RemoveObsoleteCodeBase<SyntaxKind>
+public sealed class ObsoleteAttributes : ObsoleteAttributesBase<SyntaxKind>
 {
+    internal static readonly DiagnosticDescriptor S1123 =
+        DescriptorFactory.Create(ExplanationNeededDiagnosticId, ExplanationNeededMessageFormat);
```

Comment:
The field can be inlined to the property initializer.
Most (all?) of the rule description stuff can go to the base class.

Reply:
The field is needed, so it can be referenced in ObsoleteAttributesTest.cs.
I failed to move the Descriptor to the base class, because from there the DescriptorFactory is not available.

Reply:
There is DiagnosticDescriptorFactory in the Common library. `internal static`  should also be accessible through the derived class. "Commons" has "InteralVisibleTo" to the UnitTest set so it should be accessible.